projects/showcase/cypress/support/e2e-driver.ts

```diff
@@ -1,198 +1,112 @@
-/// <reference types="cypress" />
-import Sinon from 'cypress/types/sinon';
-import pixelmatch from 'pixelmatch';
-import { PNG } from 'pngjs';
+import { PNG } from "pngjs";
+import { MapLibreAssertable, MaplibreCypressHelper } from "./maplibre-helper";
 
 export class E2eDriver {
-  private width: number;
-  private height: number;
-  private referenceImageBuffer: Buffer;
-  private alertStub: Cypress.Agent<Sinon.SinonStub<any[], any>>;
+  private helper = new MaplibreCypressHelper();
 
-  visitMapPage = (url: string): E2eDriver => {
-    cy.visit(url);
-    this.alertStub = cy.stub().as('alertStub');
-    return this;
+  beforeAndAfter = () => {
+    beforeEach(() => {});
+    afterEach(() => {});
   };
 
-  // Aliased
-  waitForMapToIdle = (timeout: number = 30000): E2eDriver => {
-    return this.when.waitForMapToIdle(timeout);
+  given = {
+    alertStub: () => this.helper.given.onAlert(this.helper.given.stub("alert")),
+    interceptStreetsSprite: () =>
+      this.helper.given.intercept("**/streets/sprite.png", "streets"),
   };
+  when = {
+    wait: (ms: number) => this.helper.when.wait(ms),
 
-  takeImageSnapshot = (): E2eDriver => {
-    this.initReferenceImage();
-    return this;
-  };
+    resetConsoleWarnings: () => this.helper.when.resetConsoleWarnings(),
+    waitForStreetsSpriteResponse: (url: string, method: string = "GET") =>
+      this.helper.when.waitForResponse("streets"),
+    waitForMapToIdle: () =>
+      this.helper.when.waitUntil(() =>
+        this.helper.get.bySelector("true", "data-idle")
```

Comment:
this method looks odd `bySelector("true", "data-idle")` I would expect it to be: bySelector("data-idle", "true")

Reply:
The second parameter has a default (data-cy).
Maybe I should add another method doing the same thing with a clearer name.